nifi/nifi-nar-bundles/nifi-flume-bundle/nifi-flume-processors/src/main/java/org/apache/nifi/processors/flume/FlumeSinkProcessor.java

```diff
@@ -0,0 +1,146 @@
+            .build();
+    public static final PropertyDescriptor SOURCE_NAME = new PropertyDescriptor.Builder()
+            .name("Sink Name")
+            .description("The name of the sink used in the Flume sink configuration")
+            .required(true)
+            .defaultValue("sink-1")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor FLUME_CONFIG = new PropertyDescriptor.Builder()
+            .name("Flume Configuration")
+            .description("The Flume configuration for the sink copied from the flume.properties file")
+            .required(true)
+            .defaultValue("")
+            .addValidator(Validator.VALID)
+            .build();
+
+    public static final Relationship SUCCESS = new Relationship.Builder().name("success").build();
+    public static final Relationship FAILURE = new Relationship.Builder().name("failure").build();
+
+    private List<PropertyDescriptor> descriptors;
+    private Set<Relationship> relationships;
+
+    private volatile Sink sink;
+    private volatile NifiSinkSessionChannel channel;
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        this.descriptors = ImmutableList.of(SINK_TYPE, AGENT_NAME, SOURCE_NAME, FLUME_CONFIG);
+        this.relationships = ImmutableSet.of(SUCCESS, FAILURE);
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @OnScheduled
+    public void onScheduled(final SchedulingContext context) {
+        try {
+            channel = new NifiSinkSessionChannel(SUCCESS, FAILURE);
+            Configurables.configure(channel, new Context());
+            channel.start();
+
+            sink = SINK_FACTORY.create(context.getProperty(SOURCE_NAME).getValue(),
+                    context.getProperty(SINK_TYPE).getValue());
+            sink.setChannel(channel);
+
+            String flumeConfig = context.getProperty(FLUME_CONFIG).getValue();
+            String agentName = context.getProperty(AGENT_NAME).getValue();
+            String sinkName = context.getProperty(SOURCE_NAME).getValue();
+            Configurables.configure(sink,
+                    getFlumeSinkContext(flumeConfig, agentName, sinkName));
+
+            sink.start();
+        } catch (Throwable th) {
+            getLogger().error("Error creating sink", th);
+            throw Throwables.propagate(th);
+        }
+    }
+
+    @OnUnscheduled
+    public void unScheduled() {
+        sink.stop();
+        channel.stop();
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context,
+            final ProcessSession session) throws ProcessException {
+
+        channel.setSession(session);
+        try {
+            if (sink.process() == Sink.Status.BACKOFF) {
```

Comment:
Ryan's view on context.yield is inline with its intent.  Generally it is a way for the processor to communicate to the framework that it doesn't want to be scheduled for a while because some transient issue is occurring that it expects will clear up 'later'.